include/multigrid_helper.cuh

```diff
@@ -8,15 +8,17 @@ namespace quda {
    */
   template <int fineSpin, int coarseSpin>
   struct spin_mapper {
-    static constexpr int spin_block_size = fineSpin / coarseSpin;
+    // fineSpin == 1, coarseSpin == 2 identifies staggered fine -> coarse w/ spin.
+    static constexpr int spin_block_size = (fineSpin == 1 && coarseSpin == 2) ? 0 : fineSpin / coarseSpin;
 
     /**
        Return the coarse spin coordinate from the fine spin coordinate
        @param s Fine spin coordinate
+       @param parity fine parity, for staggered
        @return Coarse spin coordinate
      */
-    __device__ __host__ constexpr inline int operator()( int s ) const
-    { return s / (spin_block_size > 0 ? spin_block_size : 1); }
+    __device__ __host__ constexpr inline int operator()( int s, int parity ) const
```

Comment:
This change to the spin_mapper lets me reuse existing prolongator, restrictor, and transfer code as is, agnostic to Wilson-type or staggered fermions. I just updated spin_map(s) -> spin_map(s,parity) everywhere as part of this. I don't think there'd be a Wilson-only or staggered-only kernel that'd use the spin mapper?